libraries/rush-lib/src/api/RushProjectSelector.ts

```diff
@@ -0,0 +1,183 @@
+} from '../logic/selectors/GitChangedProjectSelectorParser';
+import { NamedProjectSelectorParser } from '../logic/selectors/NamedProjectSelectorParser';
+import { TagProjectSelectorParser } from '../logic/selectors/TagProjectSelectorParser';
+import { VersionPolicyProjectSelectorParser } from '../logic/selectors/VersionPolicyProjectSelectorParser';
+import { JsonFileSelectorParser } from '../logic/selectors/JsonFileSelectorParser';
+import { SelectorError } from '../logic/selectors/SelectorError';
+import {
+  SelectorExpression,
+  IExpressionDetailedSelector,
+  ExpressionParameter,
+  IExpressionOperatorAnd,
+  IExpressionOperatorOr,
+  IExpressionOperatorNot,
+  isDetailedSelector,
+  isParameter,
+  isAnd,
+  isOr,
+  isNot
+} from './SelectorExpressions';
+
+/**
+ * When preparing to select projects in a Rush monorepo, some selector scopes
+ * require additional configuration in order to control their behavior. This
+ * options interface allows the caller to provide these properties.
+ */
+export interface IProjectSelectionOptions {
+  /**
+   * Options required for configuring the git selector scope.
+   */
+  gitSelectorParserOptions: IGitSelectorParserOptions;
+}
+
+/**
+ * A central interface for selecting a subset of Rush projects from a given monorepo,
+ * using standardized selector expressions. Note that the types of selectors available
+ * in a monorepo may be influenced in the future by plugins, so project selection
+ * is always done in the context of a particular Rush configuration.
+ */
+export class RushProjectSelector {
+  private _rushConfig: RushConfiguration;
+  private _scopes: Map<string, ISelectorParser<RushConfigurationProject>> = new Map();
+  private _options: IProjectSelectionOptions;
+
+  public constructor(rushConfig: RushConfiguration, options: IProjectSelectionOptions) {
+    this._rushConfig = rushConfig;
+    this._options = options;
+
+    this._scopes.set('name', new NamedProjectSelectorParser(this._rushConfig));
+    this._scopes.set(
+      'git',
+      new GitChangedProjectSelectorParser(this._rushConfig, this._options.gitSelectorParserOptions)
+    );
+    this._scopes.set('tag', new TagProjectSelectorParser(this._rushConfig));
+    this._scopes.set('version-policy', new VersionPolicyProjectSelectorParser(this._rushConfig));
+    this._scopes.set('json', new JsonFileSelectorParser(this._rushConfig, this));
+  }
+
+  /**
+   * Select a set of projects using the passed selector expression.
+   *
+   * The passed context string is used only when constructing error messages, in the event of
+   * an error in user input. The default string "expression" is used if no context is provided.
+   */
+  public async selectExpression(
+    expr: SelectorExpression,
+    context: string = 'expression'
+  ): Promise<RushConfigurationProject[]> {
+    if (isAnd(expr)) {
+      return this._evaluateAnd(expr, context);
+    } else if (isOr(expr)) {
+      return this._evaluateOr(expr, context);
```

Comment:
Considering that this operates in set space, `Union` and `Intersect` are probably clearer terms to use.